File: 45. Jump Game II/45. Jump Game II.dart
class Solution {
  int jump(List<int> nums) {
    if (nums.length <= 1) return 0;
    
    int jumps = 0;
    int currentMax = 0;
    int nextMax = 0;
    
    for (int i = 0; i < nums.length - 1; i++) {
      nextMax = nextMax < i + nums[i] ? i + nums[i] : nextMax;
      
      if (i == currentMax) {
        jumps++;
        currentMax = nextMax;
        
        if (currentMax >= nums.length - 1) {
          break;
        }
      }
    }
    
    return jumps;
  }
}

File: 45. Jump Game II/45. Jump Game II.py
class Solution(object):
    def jump(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if len(nums) <= 1:
            return 0
        
        jumps = 0
        current_max = 0
        next_max = 0
        
        for i in range(len(nums) - 1):
            next_max = max(next_max, i + nums[i])
            
            if i == current_max:
                jumps += 1
                current_max = next_max
                
                if current_max >= len(nums) - 1:
                    break
        
        return jumps